src/MSALWrapper/AuthFlow/AuthFlows.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.AuthFlow
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// A list of <see cref="IAuthFlow"/> to be executed by an <see cref="AuthFlowExecutor"/>.
+    /// </summary>
+    public class AuthFlows : IEnumerable<IAuthFlow>
+    {
+        /// <summary>
+        /// The Id of this group of auth flows.
+        /// </summary>
+        public readonly string Id;
+
+        private readonly IEnumerable<IAuthFlow> authFlows;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AuthFlows"/> class.
+        /// </summary>
+        /// <param name="client">The client for these <paramref name="authFlows"/>.</param>
+        /// <param name="tenant">The tenant for these <paramref name="authFlows"/>.</param>
+        /// <param name="authFlows">An <see cref="IEnumerable{IAuthFlow}"/> of auth flows.</param>
+        public AuthFlows(Guid client, Guid tenant, IEnumerable<IAuthFlow> authFlows)
+        {
+            this.Id = $"{client}_{tenant}";

```

Comment:
I am a bit unclear on why are we creating ids this way?

Reply:
@kyle-rader-msft mentioned in the description
> If they just had an Id, to name them, we could use to lock with

This matches how we're currently creating lock names, so I assume the intention is to use an ID as the lock name.

Reply:
Here, I'm specifically taking what we use today for the lock name around a set of auth flows. I could rename `Id` to LockName to make this more explicit. (In other words, I did not invent this here, it's taken from TokenFetcher)

Reply:
Aha! I see. Perhaps, we could rename it to be `LockName` and be more explicit.

Reply:
From the perspective of calling code that might make sense, but this class is unaware of any locking. It just so happens that the unique identifier for this set of auth flows can be used as the name of the lock.

Reply:
Yeah, and the fact we're exposing it indicates to me we're finding the right abstraction within AuthFlows, and that it should house AuthFlowFactory's method as a ctor, and AuthFlowExecutor as a method. Then the LockName, becomes private since it's an implementation detail. Being public here is a signal that as is, AuthFlows is a leaky abstraction. So, follow UP coming.